src/miniscript/types/extra_props.rs

```diff
@@ -553,6 +668,10 @@ impl Property for ExtData {
             ops_count_sat: ops_count_sat
                 .map(|x: usize| (x + (n - 1) + 1 + (sum + ops_count_nsat_sum as i32) as usize)), //adds and equal
             ops_count_nsat: ops_count_nsat.map(|x| x + (n - 1) + 1), //adds and equal
+            // For simplicity we only multiply the worse cost seen
```

Comment:
I think we should calculate the explicit values.
Sort by max_elem_sat - max_elem_dissat and pick the indices corresponding to max k. Add those k satisfaction costs with other `n-k` dissat costs. The same algorithm is used `max_satisfaction_witness_elements()`

Reply:
Why do you compute the OP count cost this way ?
```rust
match (sub.ops_count_sat, sub.ops_count_nsat) {
                (Some(x), Some(y)) => {
                    ops_count_sat_vec.push(Some(x as i32 - y as i32));
```

Reply:
> I think we should calculate the explicit values.

FWIW i don't think it's tractable to have an accurate value, but yeah we can have a better approximation..